### ISSUE TITLE
[RFE] CamOps Initiative Modifier

### ISSUE THREAD

**Issue by IllianiCBT** [MegaMek/mekhq#4600]
CamOps includes a lot of modifiers applied to initiative rolls made during scenarios. These can result from using a WMD or poor morale, to use the two examples that jump immediately to mind.

Due to the interactions needed with MegaMek, this falls outside my immediate skillset. So I'm putting out this RFE in the hope someone else is willing to take a shot.

### Requirements
- An initiative modifier `int` saved in `campaign` with setter and getter, and a third method that allows you to increment/decrement the modifier. This will need to be marshaled and unmarshaled to xml.
- The ability to pass that initiative modifier into a scenario and have it apply to initiative rolls made for the duration of that scenario

Implementing the various CamOps systems that would take advantage of this functionality I will be doing myself, so no need to worry about that side of things.

**Comment by Saklad5** [comment]
Based on [this ruling](https://bg.battletech.com/forums/index.php?topic=85848.msg2027228#msg2027228), MekHQ needs to communicate two things to MegaMek: the total initiative modifier, and the single highest positive modifier. For instance, if there are two +1 initiative modifiers, the total would be +2 but the highest (which acts as a cap) would be +1. In this case, a -1 initiative modifier applied by something in MegaMek would still result in an effective modifier of +1. If MegaMek applied a +2 modifier, however, the effective modifier would be +2.

TLDR: MegaMek needs an extra lobby option for the "starting cap" of initiative. The method MekHQ uses to increment the total modifier should update that starting cap if the provided increment is higher, and then pass that cap to MegaMek using the new option.

**Comment by Saklad5** [comment]
@HammerGS Was the initiative cap implemented?

**Comment by HammerGS** [comment]
> @HammerGS Was the initiative cap implemented?

I have no idea, I just push the code not write it.

**Comment by Saklad5** [comment]
> > @HammerGS Was the initiative cap implemented?
> 
> I have no idea, I just push the code not write it.

I'm not sure this should be closed, based on the pull request.

**Comment by IllianiCBT** [comment]
> @HammerGS Was the initiative cap implemented?

This would be a question for [DM0000](https://github.com/DM0000)

**Comment by DM0000** [comment]
Cap was not implemented in MehHq  -adding now.    MegaMek already had the constantInitBonus for Players which I'm using to pass the bonus value to MegaMek from MekHq.

Adding two methods to set Maxbonus -
1st regular setter
2nd  setter that checks against current bonus and takes highest

adding checks in initiativeBonus setter and incremental setter to not exceed maxBonus

**Comment by DM0000** [comment]
Correction adding applyInitBonus  that checks and sets maxBonus before applying makes more sense then a seperate maxBonus setter and checker

**Comment by Saklad5** [comment]
The max bonus may also be influenced by things in MegaMek.

**Comment by DM0000** [comment]
new PR (https://github.com/MegaMek/mekhq/pull/5260)

Would Campaign modifiers be treated the same as scenario ones?  For example does it make sense for a mekwarrior getting a 5 kill spree getting +5 bonus to carry over after  battle or be limited to the +1 modifier(ate at company kitchen)?* examples based on nothing.

Maybe a scenarioMax and campaignMax?

Thinking more about it now there would need to be a way to time out bonuses.  Maybe MaxBonus isn't even needed just array of bonuses then bonus cant be higher then highest value.  This was bonuses could be removed and old maxes remembered.  It all really relies on how Illani wants to apply the bonuses(one and done,weekly/monthly) and if they timeout.
